src/tunnel.cc

```diff
@@ -1248,18 +1263,17 @@ TunnelStateData::noteDestination(Comm::ConnectionPointer path)
 
     destinations->addPath(path);
 
-    if (usingDestination()) {
-        // We are already using a previously opened connection but also
-        // receiving destinations in case we need to re-forward.
-        Must(!transportWait);
-        return;
-    }
-
     if (transportWait) {
+        assert(!transporting());
         notifyConnOpener();
         return; // and continue to wait for tunnelConnectDone() callback
     }
 
+    // the current commitToServer()-based code cannot receive new destinations
+    // while transporting(), but future enhancements may enable this code path
```

Comment:
IMO, it is reasonable even today for the first server to be used while alternative server details keep coming in. No need to pick out this implementation detail for comment unless you want to make it a TODO.
```suggestion
    // performance optimization; avoid connecting to alternative servers once transport begins.
```
or
```suggestion
    // the current commitToServer()-based code cannot receive new destinations
    // while transporting(). TODO open and pool alternative server connections for other clients or future use.
```